app/websocket_app.rb

```diff
@@ -272,33 +275,55 @@ class WebsocketApp
     end
   end
 
+  def update_node_attributes(node_name, message)
+    node = FlightScheduler.app.default_partition.nodes.find do |n|
+      n.name == node_name
+    end
+    return unless node
+
+    attributes = node.attributes.dup
+    attributes.cpus   = message[:cpus]    if message.key?(:cpus)
+    attributes.gpus   = message[:gpus]    if message.key?(:gpus)
+    attributes.memory = message[:memory]  if message.key?(:memory)
+
+    if attributes.valid?
+      node.attributes = attributes
+    else
+      Async.logger.error <<~ERROR
+        Invalid node attributes for #{node.name}:
+        #{node.attributes.errors.messages}
+      ERROR
+    end
+  end
+
   def call(env)
     Async::WebSocket::Adapters::Rack.open(env) do |connection|
       begin
         message = connection.read
         unless message.is_a?(Hash) && message[:command] == 'CONNECTED'
           Async.logger.info("Badly formed connection message #{message.inspect}")
           connection.close
-          break
+          next
```

Comment:
Why the change here?

Reply:
I was experiencing a `LocalJumpError` when a malformed packet was being sent, which is why it went undetected for so long.

I'm not entirely sure how the internals of `sync::WebSocket::Adapters::Rack` work, but logically it makes sense to use `next` instead of `break`. In the event of an error, we want the adaptor to move to the next connection instead of breaking the process loop entirely.

Reply:
So it looks like there is no loop in the internals of `Async::WebSocket::Adapters::Rack`.  However there is a call to the provided block (`block.call`).  The `next` causes block to terminate early, whereas `break` results in an error due to the lack of a look.

Good catch @WilliamMcCumstie